### ISSUE TITLE
Ignoring duplicate file name: ... on Google Drive for Google user: ...

### ISSUE THREAD

**Issue by gvmura** [nextcloud/server#4279]
### Steps to reproduce
1. Create a Google external Storage, and assign it to group of users.
2. Upload a file from web UI or copy it from client folder.
3. Edit file and modify it, re-upload it from web UI or wait client synchronization.

### Expected behaviour
I should view the file in the web UI and in client folder.

### Actual behaviour
The file isn't viewed in web UI and the client inform that the file was removed.
On google drive there are two files with the same name and different content (updated and not).
The file appears in web UI and in client if I remove one of the two files from Google Drive UI

### Server configuration
**Operating system**: Linux nextcloud 4.4.0-72-generic #93-Ubuntu SMP Fri Mar 31 14:07:41 UTC 2017 x86_64

**Web server:** Apache/2.4.18 (Ubuntu) (apache2handler)

**Database:** mysql 5.7.17

**PHP version:** 7.0.15-0ubuntu0.16.04.4
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, apache2handler, mysqlnd, PDO, xml, apcu, bz2, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, igbinary, imagick, imap, intl, json, ldap, exif, mcrypt, mysqli, pdo_mysql, Phar, posix, readline, redis, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 11.0.2 (stable) - 11.0.2.7

**Updated from an older Nextcloud/ownCloud or fresh install:**

**Where did you install Nextcloud from:** 

**Signing status:**
<details>
	<summary>Signing status</summary>

```
[]
```

</details>

**List of activated apps:**
<details>
	<summary>App list</summary>
```
Enabled:
 - activity: 2.4.1
 - audioplayer: 1.5.1
 - comments: 1.1.0
 - dav: 1.1.1
 - encryption: 1.4.1
 - federatedfilesharing: 1.1.1
 - files: 1.6.1
 - files_antivirus: 1.0.0.0
 - files_external: 1.1.2
 - files_pdfviewer: 1.0.1
 - files_sharing: 1.1.1
 - files_texteditor: 2.2
 - files_trashbin: 1.1.0
 - files_versions: 1.4.0
 - files_videoplayer: 1.0.0
 - firstrunwizard: 2.0
 - gallery: 16.0.0
 - issuetemplate: 0.2.1
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - nextcloud_announcements: 1.0
 - notifications: 1.0.1
 - password_policy: 1.1.0
 - provisioning_api: 1.1.0
 - sharebymail: 1.0.1
 - survey_client: 0.1.5
 - systemtags: 1.1.3
 - templateeditor: 0.2
 - theming: 1.1.1
 - twofactor_backupcodes: 1.0.0
 - updatenotification: 1.1.1
 - user_ldap: 1.1.2
 - workflowengine: 1.1.1

Disabled:
 - admin_audit
 - external
 - federation
 - files_accesscontrol
 - files_automatedtagging
 - files_retention
 - serverinfo
 - user_external
 - user_saml
```
</details>

**The content of config/config.php:**
<details>
	<summary>Config report</summary>

```
{
    "instanceid": "oc6c7mckt9tu",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "nextcloud.uniss.it",
        "owncloud.uniss.it"
    ],
    "datadirectory": "\/home\/sitiweb\/nextcloud\/web\/data",
    "overwrite.cli.url": "https:\/\/nextcloud.uniss.it",
    "dbtype": "mysql",
    "version": "11.0.2.7",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "localhost",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "Europe\/Rome",
    "installed": true,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "tempdirectory": "\/home\/sitiweb\/nextcloud\/web\/tmp",
    "default_language": "it",
    "lost_password_link": "https:\/\/servizi.uniss.it\/account",
    "allow_user_to_change_display_name": false,
    "skeletondirectory": "",
    "loglevel": 0,
    "mail_from_address": "nextcloud_NOREPLY",
    "mail_smtpmode": "php",
    "mail_domain": "uniss.it",
    "ldapIgnoreNamingRules": false,
    "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
    "ldapUserCleanupInterval": 0,
    "updater.release.channel": "stable",
    "htaccess.RewriteBase": "\/",
    "filelocking.enabled": "true",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "\/var\/run\/redis\/redis.sock",
        "port": 0,
        "timeout": 0
    },
    "trashbin_retention_obligation": "14,auto",
    "maintenance": false
}
```
</details>

**Are you using external storage, if yes which one:** Array
(
    [0] => \OC\Files\Storage\Local
    [1] => \OCA\Files_External\Lib\Storage\FTP
    [2] => \OC\Files\Storage\DAV
    [3] => \OCA\Files_External\Lib\Storage\OwnCloud
    [4] => \OCA\Files_External\Lib\Storage\SFTP
    [5] => \OCA\Files_External\Lib\Storage\AmazonS3
    [6] => \OCA\Files_External\Lib\Storage\Dropbox
    [7] => \OCA\Files_External\Lib\Storage\Google
    [8] => \OCA\Files_External\Lib\Storage\Swift
    [9] => \OCA\Files_External\Lib\Storage\SFTP
)

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
	<summary>LDAP config</summary>

	```
	With access to your command line run e.g.:
	sudo -u www-data php occ ldap:show-config
	from within your Nextcloud installation folder

	Without access to your command line download the data/owncloud.db to your local
	computer or access your SQL server remotely and run the select query:
	SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


	Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
	```
</details>
### Client configuration
**Browser:** Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0

**Operating system:**

#### Web server error log
<details>
	<summary>Web server error log</summary>
	```
	Insert your webserver log here
	```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
	<summary>Nextcloud log</summary>
	```
	Insert your Nextcloud log here
	```
</details>

#### Browser log
<details>
	<summary>Browser log</summary>
	```
	Insert your browser log here, this could for example include:

	a) The javascript console log
	b) The network log
	c) ...
	```
</details>

**Comment by MorrisJobke** [comment]
cc @icewind1991

**Comment by gvmura** [comment]
I have modifyed "apps/files_external/lib/Lib/Storage/Google.php" file.
I have added at the beginning of the "rename" function the code:
`if ($this->is_file($path2)) {
	$secondoFile = $this->getDriveFile($path2);
	if ($secondoFile) {
		$this->service->files->trash($secondoFile->getId());
		\OCP\Util::writeLog('files_external', 'trash file '.$path2.' for renaming '.$path1.
		' on Google Drive.', \OCP\Util::DEBUG);
	}
}`

I have tested the version app and everything seems ok because it uses home storage to restore previous files.

The only side effect is that the trash of Google Drive isn't purged, but this happens also when a file is deleted from nextcloud with the trash app.

I am not a software developer. I tryed to solve the problem quiclky.
I apologize for my inaccuracies.
